packages/shared/lib/src/sdk/transaction.rs

```diff
@@ -32,9 +34,8 @@ pub enum TransactionKind {
 impl TransactionKind {
     pub fn from(tx_type: TxType, data: &[u8]) -> Self {
         match tx_type {
-            TxType::TransparentTransfer => TransactionKind::TransparentTransfer(
-                TransparentTransfer::try_from_slice(data)
-                    .expect("Cannot deserialize TransparentTransfer"),
+            TxType::Transfer => TransactionKind::Transfer(
+                Transfer::try_from_slice(data).expect("Cannot deserialize TransparentTransfer"),
```

Comment:
So this block of code in `shared/lib/src/sdk/tx.rs` gets the "code" section from the Tx and parses the bytes into a hash - this hash refers to the hash of the wasm used in the Tx that determines its behavior:

https://github.com/anoma/namada-interface/blob/a3754f477e7f9230d186d5e6259d9112ddd1c45a/packages/shared/lib/src/sdk/tx.rs#L170-L176 

We match these hashes to their wasm path via a query, since this can only be known after build, and can be queried from the chain storage (these checksum hashes will be different when the wasms change - they get their checksums at the end of `make build-wasm-scripts`):

https://github.com/anoma/namada-interface/blob/a3754f477e7f9230d186d5e6259d9112ddd1c45a/packages/shared/lib/src/query.rs#L776-L789 

Code path is need to issue this query, where we get a storage key from it then make a call over RPC: 

```rust
let hash_key = Key::wasm_hash(tx_code_path);
let (tx_code_res, _) = rpc::query_storage_value_bytes(&self.client, &hash_key, None, false)....
```

Here I just map the wasm "code path" to our `TxType`: https://github.com/anoma/namada-interface/blob/a3754f477e7f9230d186d5e6259d9112ddd1c45a/packages/shared/lib/src/sdk/tx.rs#L97-L123

Not sure of any documentation you could read up on Tx, but I can try to explain here: The core structure of a Tx is simple, it's just a `header` and `sections`, and the sections contain different types of data, signatures (`Authorization`), code (transaction wasm code), inner tx data, etc.:

Tx: https://github.com/anoma/namada/blob/22a4839d84b8e7ff6a909b94d50165d98434ee15/crates/tx/src/types.rs#L1032-L1049

Sections enum: https://github.com/anoma/namada/blob/22a4839d84b8e7ff6a909b94d50165d98434ee15/crates/tx/src/types.rs#L768-L788

`Header` contains things like `chain_id`, `batch`, `expiration`, and `tx_type`. `tx_type` here is different than ours, it is an enum that refers to `Raw`, `Wrapper` and `Protocol`

That's about the extent of what I understand! Still learning myself.